rotkehlchen/chain/ethereum/modules/aave/v2/decoder.py

```diff
@@ -0,0 +1,253 @@
+        user = hex_or_bytes_to_address(tx_log.topics[2])
+        return HistoryBaseEntry(
+            event_identifier=transaction.tx_hash,
+            sequence_index=tx_log.log_index,
+            timestamp=ts_sec_to_ms(transaction.timestamp),
+            location=Location.BLOCKCHAIN,
+            location_label=user,
+            event_type=HistoryEventType.INFORMATIONAL,
+            event_subtype=HistoryEventSubType.NONE,
+            asset=token,
+            balance=Balance(),
+            notes=f'{"Enable" if tx_log.topics[0] == ENABLE_COLLATERAL else "Disable"} {token.symbol} as collateral on AAVE v2',  # noqa: E501
+            counterparty=CPT_AAVE_V2,
+        )
+
+    def _decode_deposit(
+            self,
+            token: 'EvmToken',
+            tx_log: EvmTxReceiptLog,
+            decoded_events: list[HistoryBaseEntry],
+    ) -> None:
+        """Decode aave v2 deposit event"""
+        user = hex_or_bytes_to_address(tx_log.data[:32])
+        on_behalf_of = hex_or_bytes_to_address(tx_log.data[64:96]) if hex_or_bytes_to_str(tx_log.data[64:96]) != '' else None  # noqa: E501
+        amount = asset_normalized_value(
+            amount=hex_or_bytes_to_int(tx_log.data[32:64]),
+            asset=token,
+        )
+        notes = f'Deposit {amount} {token.symbol} from user {user} into AAVE v2'
+        if on_behalf_of is not None:
+            notes += f' on behalf of {on_behalf_of}'
+        for event in decoded_events:
+            if (
+                event.location_label == user and
+                event.balance.amount == amount and
+                event.event_subtype == HistoryEventSubType.NONE
+            ):
+                if event.event_type == HistoryEventType.SPEND:
+                    event.event_type = HistoryEventType.DEPOSIT
+                    event.event_subtype = HistoryEventSubType.DEPOSIT_ASSET
+                    event.notes = notes
+                    event.counterparty = CPT_AAVE_V2
+                else:
+                    event.event_subtype = HistoryEventSubType.RECEIVE_WRAPPED
+                    resolved_asset = event.asset.resolve_to_asset_with_symbol()
+                    event.notes = f'Receive {event.balance.amount} {resolved_asset.symbol} from AAVE v2'  # noqa: E501
+                    event.counterparty = CPT_AAVE_V2
+                # Set counterparty for both events
+                event.counterparty = CPT_AAVE_V2
+
+    def _decode_withdrawal(
+            self,
+            token: 'EvmToken',
+            tx_log: EvmTxReceiptLog,
+            decoded_events: list[HistoryBaseEntry],
+    ) -> None:
+        """Decode aave v2 withdrawal event"""
+        user = hex_or_bytes_to_address(tx_log.topics[2])
+        to = hex_or_bytes_to_address(tx_log.topics[3])
+        amount = asset_normalized_value(
+            amount=hex_or_bytes_to_int(tx_log.data),
+            asset=token,
+        )
+        notes = f'Withdraw {amount} {token.symbol} from AAVE v2 user {user}'
```

Comment:
The user is always the location label right? So no need to add in notes.

```suggestion
        notes = f'Withdraw {amount} {token.symbol} from AAVE v2'
```